teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md

```diff
@@ -0,0 +1,478 @@
+
+Tekton Chains will follow [TEP 0075](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md) and [TEP 0076](https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md) and support the sets of json schemas and described in the documents. In this proposal, the expected formats of params and results are also defined.
+
+
+### Notes and Caveats
+
+When these well defined structures grow as richer provenance are pursued, the size of the structure can eventually grow beyong the limit of the container termination message. [TEP 0086](https://github.com/tektoncd/community/pull/521) is addressing this issue.
+
+
+## Design Details
+
+### Signable Targets
+
+All signable targets should be provided in a structure that is well defined in the proposal. If any field is missed, the signable targets will be skipped and an error will be thrown.
+
+#### Version Control Sources
+
+Based on the use cases and data storage within provenance described above, it makes sense to have the source object to be the following because all version control sources will be grinded down into the requirements in the Predicate’s materials:
+```
+  ...
+  - name: CHAINS-VCS
+    type: object
+    description: |
+      The source distribution
+      * uri: resource uri of the git commit
+      * digest: revision digest
+      * alg: digest algorithm: sha256 etc.
+      * src_type: source VCS type: git, hg, etc.
+    properties:
+    src_type:
+      type: string
+    uri:
+      type: string
+    alg:
+      type: string
+    digest:
+      type: string
+```
+The required formats for src_type is one of `git`, `hg`, `svc`, `baz`, `cvs`, `perforce`. The source types can be easily extended when more VCS are supported. The required formats for uri should follow [ResourceURI](https://github.com/in-toto/attestation/blob/main/spec/field_types.md#field-type-definitions). The required format for alg is one of pre-defined algorithm names in intoto attestation [field type definitions](https://github.com/in-toto/attestation/blob/main/spec/field_types.md#field-type-definitions). The digest is the revision digest.
+
+#### OCI Images
+
+OCI image sources will have the following schema:
+```
+...
+  - name: CHAINS-OCI_IMAGE
+    type: object
+    description: |
+      The source distribution
+      * name: name of the built image
+      * digest: image digest
+      * alg: digest algorithm: sha256 etc.
+    properties:
+    name:
+      type: string
+    digest:
+      type: string
+    alg:
+      type: string
+```
+The `name` is the OCI image name, for example, `gcr.io/test/foo`. The `digest` will be the image digest in the form of `69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f`. The `alg` follows the same requirements for VCS `alg` field.
```

Comment:
Same as above, this is really just standardizing on SLSA's `subject` field, right?

Reply:
Yep!